vaadin-flow-components-shared-parent/vaadin-flow-components-base/src/main/java/com/vaadin/flow/component/shared/ValidationError.java

```diff
@@ -0,0 +1,17 @@
+package com.vaadin.flow.component.shared;
+
+import java.io.Serializable;
+
+public final class ValidationError implements Serializable {
+    public static final String REQUIRED = "VALIDATION_REQUIRED_ERROR";
```

Comment:
Make sure those non-explaining error messages are not overriding errors provided by the binder (like asRequired("My awesome name of the field is empty"). 

Additionally, if those strings are ever shown to the end-user there should be a way to translate them. All fields should provide an i18n mapping for the possible errors and they outcome.

Reply:
> Make sure those non-explaining error messages are not overriding errors provided by the binder (like asRequired("My awesome name of the field is empty").

Hmmm. That's a valid point. Currently the `HasValidator` with `Binder` works so that the `getDefaultValidator()` is called before the validation constraints from `Binder` are called. So if `asRequired()` is called on the binding definition, that will effectively set the component's `required` property to `true` and the validation will fail with the `VALIDATION_REQUIRED_ERROR` message. From `getDefaulltValidator()`, I can't see a way of checking if there's a custom message defined on `asRequired()`, though. 
I also don't think we can't remove the required check from the component (even when used with `Binder`), since my assumption is that it should work if the required is set on the component `setRequired()/`setRequiredIndicatorVisible()`. So I need to investigate how to proceed here.

> Additionally, if those strings are ever shown to the end-user there should be a way to translate them. All fields should provide an i18n mapping for the possible errors and their outcome.

Good point. There should be a way of defining custom messages.

Reply:
Just a thought: I don't think it's possible to fix this behavior on your side if getDefaultValidator runs before the binder validations. I think this should be fixed / done inside flow - and the getDefaultValidator() should run AFTER the binder validation; meaning the getDefaultValidator() should be registered once the field and it's validations are bind. This allows to stop the validations once the binder fails and uses this message instead of the messages / validations provided later by the field

Reply:
We decided to exclude required from the validation done by `getDefaultValidator()`, so if one wants to use binder, then only `asRequired()` should be used.

Reply:
That races multiple contains.. short question: Can the field validation be disabled? We have moved all our validations to the binder over the years, because the client side wasn't  secure and intuitive.

Just another example we currently use:
 - DatePicker:.setMax(LocalDate.now()); is used to disable to get the user instant visible feedback that this date isn't selectable, but the user can still type tomorrow in.. resulting on a validation on the server with a proper error message like "Birthday can't be set to the future." 
 
 With this change, what error message is shown? The server side error message or the field validation? I would expect the later.

Reply:
First, thanks for all the feedback so far, they have been raising some internal conversation about the feature.

For now, we want to build a MVP to make this behaviour change on the fields to make client and flow constraints part of the binder validation. Later on, we will make the adjustments needed, like the ones you suggested.

With that said, we have been considering have an API to disable component validation and also to deliver these changes under a feature flag, so that we can release it and validate the new behaviour with more customers before making it the default behaviour.

Reply:
> deliver these changes under a feature flag

This would be really good! Thanks for the discussion! 

Just to clarify: I'm totally with the addition of the feature, because I've requested this as "bug" 2 years ago (https://github.com/vaadin/flow/issues/7767) - but it feels the currently MVP is behind the feature richness of the binder in terms of user and developer experience, that's why I wanted to mention it. Additionally because of the issue mentioned, we have migrated all our Vaadin 8 applications to the Binder only paradigm in V23.1 - where it would be a huge effort to adopt to a surprising change like this. 

Something similar happened to us with an addtional Binder feature / behaviour change in the past, where it did something we didn't expect and had problems with the behavior change, because we heavily heavily really on Binder as the stable foundation of all our applications (validation and security) - https://github.com/vaadin/flow/issues/10342